src/app/admin/admin.component.ts

```diff
@@ -0,0 +1,121 @@
+   *
+   * @readonly
+   * @type {boolean}
+   * @memberof AdminComponent
+   */
+  get initialized(): boolean {
+    return !this._initialized.includes(false);
+  }
+
+  ngOnInit() {
+    // hide CLARK navbar
+    this.navbarService.hide();
+
+    // set the can scroll value to determine whether or not we add 30px of padding to the bottom of the content wrapper
+    const canScroll = this.route.snapshot.firstChild.data.canScroll;
+    this.canScroll = typeof canScroll === 'boolean' ? canScroll : true;
+
+    // listen for route changes to perform the same check above
+    this.router.events.pipe(
+      filter(event => event instanceof NavigationEnd),
+      map(() => this.route),
+      map(route => route.firstChild),
+      switchMap(route => route.data),
+    ).subscribe(data =>  {
+      this.canScroll = typeof data.canScroll === 'boolean' ? data.canScroll : true;
+    });
+
+    if (this.authService.hasEditorAccess()) {
+      this.editorMode = true;
+      // we don't need to load collections, so we can set that initialization block to true
+      this._initialized[0] = true;
+    } else {
+      // this user isn't an admin/editor so fetch the route parameter representing the selected collection
+      this.route.paramMap
+        .pipe(takeUntil(this.destroyed$))
+        .subscribe(params => {
+          // we don't need to authorize here since that's done in the route guard
+          this.activeCollection = params.get('collection');
+        });
+
+      this.retrieveAuthorizedCollections().then(() => {
+        // collections array is now set and we should redirect to the first collection
+        if (!this.activeCollection) {
+          this.router.navigate([this.authorizedCollections[0].abvName], { relativeTo: this.route });
+        }
+      }).catch(error => {
+        this.toaster.notify('Error!', 'There was an error retrieving collections. Please try again later.', 'bad', 'far fa-times');
+        console.error(error);
+      });
+    }
+  }
+
+  /**
+   * Retrieve a list of collection names for which the user is authorized to view (has curator status for)
+   *
+   * @memberof AdminComponent
+   */
+  async retrieveAuthorizedCollections() {
+    // wait for user to be logged in (edge case) and then fetch list of curated collections
+    return await this.authService.isLoggedIn.pipe(
+      skipWhile(x => x === false),
+      take(1)
+    )
+    .toPromise()
+    .then(async () => {
+      // we're sure the user is logged in here and so access groups should be defined
+      return await Promise.all(
+        this.authService.user['accessGroups']
+        .filter(group => group.includes('curator@'))
+        .map(group =>
+          this.collectionService.getCollection(group.split('@')[1]).then(c => this.authorizedCollections.push(c))
+        ))
+        .then(() => {
+          // remove the initialization block
+          this._initialized[0] = true;
+        });
```

Comment:
I'm not sure if I'm fully following here. What I was thinking was that the anonymous function being used by the top level `then` (which responds to the promisified data from the pipe), is just one statement - a `Promise.all`. That statement though does quite a bit, and as an uniformed reader it's a little hard for me to grasp it.

My thought was that we could create a private helper method and pass that to the top level `then` that is used. So it would read:
```
.then(doTheThing)
```

This would allow for us to put JSDoc on that function to explain the purpose of the filter, mapping, and cleanup logic of the nested `then`.

Reply:
At that point though, isn't the parent function just waiting for the user to be logged in and then handing off to another function? It's not handling any of the actual logic for the use case.

Reply:
Yes, that's correct. It would be setting things up and then invoking another function. Functions are a unit of decomposition that should be much smaller than a use case. We don't actually ever **need** a function, but they can significantly help in the way that our minds process code by reducing scope.

My suggestion here is just rooted in an opinion that promises in promises are not inherently readable - it reminds me of callback hell. I do think it would be best to decompose it, but since we don't have any guidelines on this as a team I can't say that I believe we must break it down.

I'll leave it up to your judgement.